### PR TITLE
Don't autosave unchanged has_one through records

### PR DESCRIPTION
**Pull request by alan**
The bug is with the autosave logic. It was calling save on all has_one through records in the callbacks so that meant that the after_commit callback gets called on the has_one through records even though they don't participate in the transaction!

### CODE REVIEW

**Comment by alan** [comment]
@Lolecule want to double check? Opened a PR to my own repo so that it could reviewed before sending it to Rails.

cc/ @greysteil @jamienglish

**Comment by SteveParrington** [comment]
Yeah it looks good to me, hopefully they'll get it merged soon!

File: activerecord/test/cases/autosave_association_test.rb [hunk 1]
```diff
@@ -1116,6 +1119,22 @@ def test_should_not_load_the_associated_model
...
+  self.use_transactional_fixtures = false unless supports_savepoints?
+
+  def setup
+    super
+    organization = Organization.create
+    @member = Member.create
+    MemberDetail.create(organization: organization, member: @member)
+  end
+
+  def test_should_not_save_organization
```
Comment (english) [inline activerecord/test/cases/autosave_association_test.rb]: The other test cases don't seem to mention the specifics of the instances, but rather the general purpose of the test, so maybe something along the lines of `test_should_not_save_has_one_through_model` would fit better?

Comment (alan) [inline activerecord/test/cases/autosave_association_test.rb]: Good catch!

**Comment by greysteil** [comment]
@alan - you should close this, it's on your own repo :)

**Pull request closed without merging**